reactive/kubernetes_master.py

```diff
@@ -2310,3 +2339,15 @@ def get_dns_provider():
 
     leader_set(auto_dns_provider=dns_provider)
     return dns_provider
+
+
+@when('config.changed.loadbalancer-ips')
+def vip_changed():
+    # get a new cert
+    if (is_state('certificates.available') and
+            is_state('kube-api-endpoint.available')):
+        send_data()
+
+    # update workers
+    if (is_state('kube-api-endpoint.available')):
+        push_service_data()
```

Comment:
No need for this handler. `send_data` and `push_service_data` run on every hook invocation anyway.

Reply:
Wouldn't this handler prevent a "change the config and wait 5 minutes for a hook" or are hooks guaranteed to run on config change?

Reply:
When the config changes, Juju calls the charm's `config-changed` hook, which runs reactive, which runs any handlers whose decorator conditions are met. Both `send_data` and `push_service_data` have conditions that lead them to run repeatedly, so in this case, they will run during the `config-changed` hook and use the new config value. No 5 min delay.